src/test/rgw/test_ratelimit_unittest.cc

```diff
@@ -0,0 +1,191 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include <gtest/gtest.h>
+#include "rgw/rgw_qos.h"
+
+
+using namespace std::chrono_literals;
+
+TEST(RGWRateLimit, shouldratelimit1)
+{
+  // info.enabled = false, so no limit
+  std::unique_ptr<CephContext> cct = std::make_unique<CephContext>(CEPH_ENTITY_TYPE_ANY);
+  std::shared_ptr<ActiveRateLimiter> qos(new ActiveRateLimiter(cct.get()));
+  qos->start();
+  auto active = qos->get_active();
```

Comment:
since this is unit testing `qos_entry` specifically, we shouldn't need to deal with `ActiveRateLimiter` or `RateLimiter` at all. that's what i was getting at with my earlier [comment](https://github.com/ceph/ceph/pull/42891/#discussion_r719708380):

> i'd really like to see this factored out into its own class so we can write unit tests against it

that would mean splitting the counter logic out of `RateLimiter` and into functions on `qos_entry` (either as member functions, or free functions taking `qos_entry&`). so ideally, we'd have `class ActiveRateLimiter` that only deals with the GC, `class RateLimiter` that only deals with the map of entries, and `struct qos_entry` that only deals with the counters

that way these unit tests can ignore `ActiveRateLimiter`/`RateLimiter` entirely, unless we decide to add test cases that exercise those too